packages/gatsby-plugin-sass/src/gatsby-node.js

```diff
@@ -1,64 +1,57 @@
-  const cssModulesConfDev = `${cssModulesConf}&sourceMap&localIdentName=[name]---[local]---[hash:base64:5]`
+exports.modifyWebpackConfig = ({ boundActionCreators, stage, rules, plugins, loaders }) => {
+  const { setWebpackConfig } = boundActionCreators
+  const PRODUCTION = stage !== `develop`
 
-  switch (stage) {
-    case `develop`: {
-      config.loader(`sass`, {
-        test: /\.s(a|c)ss$/,
-        exclude: /\.module\.s(a|c)ss$/,
-        loaders: [`style`, `css`, `sass`],
-      })
+  const sassLoader = {
+    loader: require.resolve(`sass-loader`),
+    options: { sourceMap: !PRODUCTION },
+  }
 
-      config.loader(`sassModules`, {
-        test: /\.module\.s(a|c)ss$/,
-        loaders: [`style`, cssModulesConfDev, `sass`],
-      })
-      return config
-    }
-    case `build-css`: {
-      config.loader(`sass`, {
-        test: /\.s(a|c)ss$/,
-        exclude: /\.module\.s(a|c)ss$/,
-        loader: ExtractTextPlugin.extract([`css?minimize`, `sass`]),
-      })
+  const sassRule = {
+    test: /\.s(a|c)ss$/,
+    exclude: /\.module\.s(a|c)ss$/,
+    use: plugins.extractText.extract({
+      fallback: loaders.style,
+      use: [loaders.css({ importLoaders: 1 }), loaders.postcss(), sassLoader],
+    }),
+  }
+  const sassRuleModules = {
+    test: /\.module\.s(a|c)ss$/,
+    use: plugins.extractText.extract({
+      fallback: loaders.style,
+      use: [
+        loaders.css({ modules: true, importLoaders: 1 }),
+        loaders.postcss(),
+        sassLoader,
+      ],
+    }),
+  }
 
-      config.loader(`sassModules`, {
-        test: /\.module\.s(a|c)ss$/,
-        loader: ExtractTextPlugin.extract(`style`, [cssModulesConf, `sass`]),
-      })
-      return config
-    }
-    case `build-html`: {
-      config.loader(`sass`, {
-        test: /\.s(a|c)ss$/,
-        exclude: /\.module\.s(a|c)ss$/,
-        loader: `null`,
-      })
+  let configRules = []
 
-      config.loader(`sassModules`, {
-        test: /\.module\.s(a|c)ss$/,
-        loader: ExtractTextPlugin.extract(`style`, [cssModulesConf, `sass`]),
-      })
-      return config
-    }
-    case `build-javascript`: {
-      config.loader(`sass`, {
-        test: /\.s(a|c)ss$/,
-        exclude: /\.module\.s(a|c)ss$/,
-        loader: `null`,
-      })
+  switch (stage) {
+    case `develop`:
+    case `build-css`:
+    case `build-javascript`:
+      configRules = configRules.concat([sassRule, sassRuleModules])
+      break
 
-      config.loader(`sassModules`, {
-        test: /\.module\.s(a|c)ss$/,
-        loader: ExtractTextPlugin.extract(`style`, [cssModulesConf, `sass`]),
-      })
-      return config
-    }
-    default: {
-      return config
-    }
+    case `build-html`:
+    case `develop-html`:
+      configRules = configRules.concat([
+        {
+          ...sassRule,
+          use: loaders.null,
+        },
+        sassRuleModules,
+      ])
+      break
   }
+
+  setWebpackConfig({
+    module: {
+      rules: configRules,
+    },
+  })
```

Comment:
I feel like overall this API isn't the best but I wanted to let folks handle merging themselves if needed. That is a bit tough with out the api-runnner works, because it doesn't really compose results together (and webpack-merge doesn't mutate the original). Sticking the config in redux is overkill, considering it only really exists there to allow mutation and replacement by plugins.

A simpler api would be to have the api runner pass the updated config from the previous plugin to the next one, but that was a bit weird to make happen.